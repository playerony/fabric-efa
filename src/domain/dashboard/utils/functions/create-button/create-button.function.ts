export function createButton(id: string): HTMLButtonElement {
  const buttonElement = document.createElement('button');

  buttonElement.id = id;
  buttonElement.innerText = 'Change color';

  buttonElement.style.zIndex = '2000';
  buttonElement.style.position = 'absolute';

  return buttonElement;
}
