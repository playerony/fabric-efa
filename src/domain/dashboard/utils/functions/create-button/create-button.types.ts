export interface Input {
  id: string;
}
