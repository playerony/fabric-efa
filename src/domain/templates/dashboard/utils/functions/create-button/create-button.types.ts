export interface Input {
  id: string;
  width: number;
  height: number;
  canvasElementX: number;
  canvasElementY: number;
}
