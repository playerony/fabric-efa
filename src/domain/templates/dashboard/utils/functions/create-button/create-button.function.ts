import { Input } from './create-button.types';

export function createButton({
  id,
  width,
  height,
  canvasElementX,
  canvasElementY,
}: Input): HTMLButtonElement {
  const buttonElement = document.createElement('button');

  const originX = width / 2;
  const originY = height / 2;
  const buttonWidth = `${width}px`;
  const buttonHeight = `${height}px`;
  const positionX = `${canvasElementX - originX}px`;
  const positionY = `${canvasElementY - originY}px`;

  buttonElement.id = id;
  buttonElement.innerText = 'Change color';

  buttonElement.style.zIndex = '25';
  buttonElement.style.cursor = 'pointer';
  buttonElement.style.position = 'absolute';
  buttonElement.style.textTransform = 'uppercase';

  buttonElement.style.top = positionY;
  buttonElement.style.left = positionX;
  buttonElement.style.width = buttonWidth;
  buttonElement.style.height = buttonHeight;

  return buttonElement;
}
