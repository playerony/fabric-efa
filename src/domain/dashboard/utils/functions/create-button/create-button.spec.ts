import { createButton } from './create-button.function';
import { functionImportTest, toMatchSnapshot } from '@utils';

function createSnapshotTestMock() {
  const width = 100;
  const height = 30;
  const canvasElementX = 500;
  const canvasElementY = 500;
  const id = 'test_button_id';

  return createButton({ id, width, height, canvasElementX, canvasElementY });
}

describe('createButton Function', () => {
  functionImportTest(createButton);
  toMatchSnapshot(createSnapshotTestMock);
});
