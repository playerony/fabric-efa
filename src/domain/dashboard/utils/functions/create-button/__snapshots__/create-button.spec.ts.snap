// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createButton Function snapshot test should match snapshot 1`] = `
<button
  id="test_button_id"
  style="z-index: 2000; cursor: pointer; position: absolute; text-transform: uppercase; top: 485px; left: 450px; width: 100px; height: 30px;"
/>
`;
